tests/v3_api/test_secrets.py

```diff
@@ -0,0 +1,338 @@
+
+    print "Verify that the secret does not exist " \
+          "and the error code returned is non zero "
+    if result != 0:
+        assert True
+
+
+def create_and_validate_workload_with_secret_as_volume(p_client, secret, ns,
+                                                       keyvaluepair):
+    name = random_test_name("test")
+    # Create Workload with secret as volume
+    mountpath = "/test"
+    volumeMounts = [{"readOnly": False, "type": "volumeMount",
+                     "mountPath": mountpath, "name": "vol1"}]
+    con = [{"name": "test1",
+            "image": "sangeetha/testclient",
+            "volumeMounts": volumeMounts}]
+
+    secretId = secret['name']
+
+    volumes = [{"type": "volume", "name": "vol1",
+                "secret": {"type": "secretVolumeSource", "defaultMode": 256,
+                           "secretId": secretId,
+                           "optional": False, "items": "NULL"}}]
+
+    workload = p_client.create_workload(name=name,
+                                        containers=con,
+                                        namespaceId=ns.id, volumes=volumes)
+    validate_workload_with_secret(p_client, workload, "deployment",
+                                  ns.name, keyvaluepair,
+                                  workloadwithsecretasVolume=True)
+
+    return workload
+
+
+def create_and_validate_workload_with_secret_as_env_variable(p_client, secret,
+                                                             ns, keyvaluepair):
+    name = random_test_name("test")
+    # Create Workload with secret as env variable
+    secretId = secret['name']
+
+    environmentdata = [{
+        "source": "secret",
+        "sourceKey": None,
+        "sourceName": secretId
+    }]
+    con = [{"name": "test",
+            "image": "sangeetha/testclient",
+            "environmentFrom": environmentdata}]
+
+    workload = p_client.create_workload(name=name,
+                                        containers=con,
+                                        namespaceId=ns.id)
+    validate_workload_with_secret(p_client, workload, "deployment",
+                                  ns.name, keyvaluepair,
+                                  workloadwithsecretasenvvar=True)
+    return workload
+
+
+def create_secret(keyvaluepair, singlenamespace=False):
+
+    p_client = namespace["p_client"]
+    name = random_test_name("default")
+    ns = namespace["ns"]
+    keyvaluepair = {"testall": "valueall"}
+
+    if not singlenamespace:
+        secret = p_client.create_secret(name=name, namespaceId='NULL',
+                                        stringData=keyvaluepair)
+        assert secret['baseType'] == "secret"
+    else:
+        secret = p_client.create_namespaced_secret(name=name,
+                                                   namespaceId=ns['name'],
+                                                   stringData=keyvaluepair)
+        assert secret['baseType'] == "namespacedSecret"
+
+    print secret
```

Comment:
can we move this secret validation as common method that can be used by update secret method as well ? Also check for all key value pair instead of just 1 item